### PR TITLE
ENG-4992 The CM must be able to select the ingress path to use for the plugin health check based on a plugin descriptor's property

### PR DESCRIPTION
**Pull request by MEM2677**

### CODE REVIEW

File: src/main/java/org/entando/kubernetes/model/bundle/descriptor/plugin/PluginDescriptor.java [hunk 1]
```diff
@@ -64,6 +64,10 @@ public static class DescriptorMetadata {
      * The relative path of the standard kubernetes health check.
      */
     private String healthCheckPath;
+    /**
+     * Generate the ingress path for the helathcheck endpoint using either the canonical ingress or the custom one.
```
Comment (firegloves) [inline src/main/java/org/entando/kubernetes/model/bundle/descriptor/plugin/PluginDescriptor.java]: please correct this typo from helathcheck to healthcheck

File: src/main/java/org/entando/kubernetes/model/bundle/installable/PluginInstallable.java [hunk 2]
```diff
@@ -35,32 +37,35 @@ public CompletableFuture<Void> install() {
             logConflictStrategyAction();
 
             EntandoPlugin plugin = BundleUtilities.generatePluginFromDescriptor(representation);
+            // ENG-4990
```
Comment (firegloves) [inline src/main/java/org/entando/kubernetes/model/bundle/installable/PluginInstallable.java]: can we remove this comment?

File: src/main/java/org/entando/kubernetes/model/bundle/installable/PluginInstallable.java [hunk 3]
```diff
@@ -35,32 +37,35 @@ public CompletableFuture<Void> install() {
...
+            boolean useCanonicalIngressPath = representation.getHealthCheckIngress() != null
+                    && representation.getHealthCheckIngress().equals(HEALTHCHECK_INGRESS_TYPE_CANONICAL);
```
Comment (firegloves) [inline src/main/java/org/entando/kubernetes/model/bundle/installable/PluginInstallable.java]: please move these lines into a method expressing what you are trying to achieve, for example `isCanonicalIngressPath()`

File: src/main/java/org/entando/kubernetes/validator/descriptor/PluginDescriptorValidator.java [hunk 4]
```diff
@@ -216,6 +220,28 @@ private PluginDescriptor validateCustomIngressPathOrThrow(PluginDescriptor descr
...
+    /**
+     * Validate the choice and assign a default, if needed.
+     *
+     * @param descriptor the plugin descriptor to validate
+     * @return InvalidBundleException if the value is not "canonical" or "custom" (case-insensitive)
+     */
+    private PluginDescriptor validateIngressHealthChoiceOrThrow(PluginDescriptor descriptor) {
+        if (StringUtils.isEmpty(descriptor.getHealthCheckIngress())) {
+            descriptor.setHealthCheckIngress(HEALTHCHECK_INGRESS_TYPE_DEFAULT);
+        } else {
```
Comment (firegloves) [inline src/main/java/org/entando/kubernetes/validator/descriptor/PluginDescriptorValidator.java:232]: I think we could remove this `else` to simplify the code without breaking the flow, am I right?

**Review by firegloves — CHANGES_REQUESTED** [review]

File: src/main/java/org/entando/kubernetes/model/bundle/descriptor/plugin/PluginDescriptor.java (hunk 1 above)
Comment (MEM2677) [reply src/main/java/org/entando/kubernetes/model/bundle/descriptor/plugin/PluginDescriptor.java]: Done!

File: src/main/java/org/entando/kubernetes/model/bundle/installable/PluginInstallable.java (hunk 2 above)
Comment (MEM2677) [reply src/main/java/org/entando/kubernetes/model/bundle/installable/PluginInstallable.java]: Done!

File: src/main/java/org/entando/kubernetes/model/bundle/installable/PluginInstallable.java (hunk 3 above)
Comment (MEM2677) [reply src/main/java/org/entando/kubernetes/model/bundle/installable/PluginInstallable.java]: Done!

File: src/main/java/org/entando/kubernetes/validator/descriptor/PluginDescriptorValidator.java (hunk 4 above)
Comment (MEM2677) [reply src/main/java/org/entando/kubernetes/validator/descriptor/PluginDescriptorValidator.java:232]: Correct, but if the first condition is met then performing validation tests is not necessary and the exception cannot be thrown; I'd leave it as is just for sake of efficiency,  but you just tell me :)

**Review by firegloves — APPROVED** [review]
LGTM

**Pull request merged**